### ISSUE TITLE
Operations like rewrite, split shouldn't require a clean state

### ISSUE THREAD

**Issue by hftf** [git-up/GitUp#433]
It would be better to just stash the changes (and tell the user that they've been stashed to prevent confusion – or alternatively, ask for permission to stash) and then unstash after the operation succeeds/fails.

**Comment by aviflax** [comment]
I agree, and I’ve also found cases wherein GitUp says an operation can’t be performed when the repo isn’t clean, but the git CLI *is* able to perform the operation. I don’t know, maybe the git CLI is using this stash/unstash approach, I don’t know, but if the CLI can do it, I’d think/hope that GitUp can do it.
